runtime/lua/vim/lsp/buf.lua

```diff
@@ -495,6 +495,69 @@ function M.outgoing_calls()
   call_hierarchy(ms.callHierarchy_outgoingCalls)
 end
 
+--- @param type_hierarchy_items lsp.TypeHierarchyItem[]?
+local function pick_type_hierarchy_item(type_hierarchy_items)
+  if not type_hierarchy_items then
+    return
+  end
+  if #type_hierarchy_items == 1 then
+    return type_hierarchy_items[1]
+  end
+  local items = {}
+  for i, item in pairs(type_hierarchy_items) do
+    local entry = item.detail or item.name
+    table.insert(items, string.format('%d. %s', i, entry))
+  end
+  local choice = vim.fn.inputlist(items)
```

Comment:
This could use `vim.ui.select` instead. It's slightly easier to use and provides a more consistent interface for users, as the other lsp operations also use it.